ext/pdo_pgsql/pgsql_driver.c

```diff
@@ -1341,7 +1415,7 @@ static int pdo_pgsql_handle_factory(pdo_dbh_t *dbh, zval *driver_options) /* {{{
 		goto cleanup;
 	}
 
-	PQsetNoticeProcessor(H->server, (void(*)(void*,const char*))_pdo_pgsql_notice, (void *)&dbh);
+	PQsetNoticeProcessor(H->server, (void(*)(void*,const char*))_pdo_pgsql_notice, (void *)dbh);
```

Comment:
nit: not a huge fan of the forced cast here, granted you did not do it but the cast of the db handler should be done within the callback IMHO.

Reply:
I'm personally inclined to keep the declaration explicitely typed (to make clear what is expected as a parameter),
but I see that multiple exts tend toward the `void *` declaration you advocate for (looking for "callback" in firebird_driver.c, pdo_sqlite.c, libmagic/apprentice.c, intl/converter/converter.c, … and this last one is curiously opposed to intl/uchar/uchar.c where the callback is declared fully typed then forced cast when passed to the lib).

So yes, let us unify pdo_pgsql.

Reply:
It s mainly because it is the proper type

```c
typedef void (*PQnoticeProcessor) (void *arg, const char *message);
...
extern PQnoticeProcessor PQsetNoticeProcessor(PGconn *conn,
                                                                                              PQnoticeProcessor proc,
											  void *arg);
```